tapis-project/tapis-ui
# task/TUI-91 - react query login hook

## Overview: ##

Demonstration of React-Query based hooks for login

## Related Github Issues: ##

* [TUI-91](https://github.com/tapis-project/tapis-ui/issues/91)

## Summary of Changes: ##

- Added `TapisProvider` context for specifying `tenantUrl` and caching auth information. This replaces `config` objects per component
- Added tapis-hooks with `authenticator/useLogin` hook
- Modified tapis-ui/Login component to have `onAuth` and `onError` callbacks and use the new `useLogin` hook

## Testing Steps: ##
1. npm install and npm run dev
2. Log in with an invalid username and password. You should see an error message in the login screen.
3. Log in with a valid username and password. The login screen will redirect to the dashboard, but it will still not show as logged in. This is expected, since the dashboard is still reading tapis-redux for login state.

## UI Photos:

## Notes: ##
